src/goto-instrument/contracts/contracts.cpp

```diff
@@ -665,24 +679,37 @@ void code_contractst::instrument_call_statement(
 
   if(called_function == "malloc")
   {
-    // malloc statments return a void pointer, which is then cast and assigned
-    // to a result variable. We iterate one line forward to grab the result of
-    // the malloc once it is cast.
-    instruction_it++;
-    if(instruction_it->is_assign())
+    const auto &function_call =
+      to_code_function_call(instruction_it->get_code());
+    if(function_call.lhs().is_not_nil())
     {
-      instrument_assign_statement(instruction_it, body, assigns);
-      const exprt &lhs = instruction_it->assign_lhs();
-      assigns.add_to_write_set(dereference_exprt(lhs));
+      const auto &object = function_call.lhs();
+
+      instruction_it++;
```

Comment:
Why do we need to iterate here? Could you add a comment?

Reply:
Added a comment here.